### PR TITLE
NIAD-1195: Make name and identifier properties optional on PerformerNameAndAddress

### PR DESCRIPTION
**Pull request by n-r-b**
## Description

Modified PerformerNameAndAddress to support both Performing Organisations with only an identifier and code (NHS002), or only a name (NHS003).

NHS002 example:
```
S01+01'
NAD+SLA+REF00:903'
SPR+ORG'
```

NHS003 example:
```
S01+01'
NAD+SLA+++ST JAMES?'S UNIVERSITY HOSPITAL'
SPR+ORG'
```

Updated integration tests with NHS002 test edifact file containing `NAD+SLA`.

## Jira Ticket

https://gpitbjss.atlassian.net/browse/NIAD-1195

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [x] Acceptance Criteria met
- [x] Commit messages are meaningful
- [x] Manually tested
- [x] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [x] If your pull request depends on any other, please link them in the description
- [ ] main branch is passing/green on CI
- [x] Add/update any relevant documentation

### CODE REVIEW

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddress.java [hunk 1]
```diff
@@ -49,27 +52,23 @@ public static PerformerNameAndAddress fromString(String edifactString) {
...
+        HealthcareRegistrationIdentificationCode code = null;
+        if (colonSplit.length > 1 && StringUtils.isNotBlank(colonSplit[0])) {
+            identifier = colonSplit[0];
+            code = StringUtils.isNotBlank(colonSplit[1])
+                ? HealthcareRegistrationIdentificationCode.fromCode(colonSplit[1]) : null;
         }
 
-        String code = colonSplit[1];
-        String practitionerName = keySplit[4];
+        final String name = keySplit.length > 3 && StringUtils.isNotBlank(keySplit[4]) ? keySplit[4] : null;
```
Comment (andrewwa-kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddress.java:66]: #94 introduces `arrayGetSafe` which we could use here, with some tweaking if you want to keep the conditional logic about only reading a code if there's an identifier:
```
        var identifier = arrayGetSafe(colonSplit, 0).orElse(null);
        var code = arrayGetSafe(colonSplit, 1)
            .map(HealthcareRegistrationIdentificationCode::fromCode)
            .orElse(null);
        var name = arrayGetSafe(keySplit, 4).orElse(null);
```

Comment (andrewwa-kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddress.java:66]: Off-by-one errors are fun!

If `array.length == 4` then `array[4]` doesn't exist.
```suggestion
        final String name = keySplit.length > 4 && StringUtils.isNotBlank(keySplit[4]) ? keySplit[4] : null;
```

File: src/intTest/resources/success_uat_data/NHS002/full_acceptance_without_nhsack.edifact.dat [hunk 2]
```diff
@@ -0,0 +1,28 @@
+UNA:+.? '
+UNB+UNOB:2+110000044300001:80+110002063600005:80+030304:2347+428090++MEDRPT'
+UNH+000001+MEDRPT:0:1:RT:NHS002'
+BGM+LSR'
+DTM+137:200303042347:203'
+S01+01'
+NAD+SLA+++TRELISKE HAE/CH'
+SPR+DPT'
+S01+01'
+NAD+SLA+REF00:902'
```
Comment (n-r-b) [inline src/intTest/resources/success_uat_data/NHS002/full_acceptance_without_nhsack.edifact.dat:10]: This is actually `NAD+SLA+REF00:903'` in the edifact test files, but `HealthcareRegistrationIdentificationCode` `903` does not currently exist

File: src/intTest/resources/success_uat_data/NHS002/full_acceptance_without_nhsack.edifact.dat [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+S06+06'
+RFF+SSI:C361380'
+ADR++US:THE WINDMILL:WINDMILL HILL++PL1 1AA'
+S07+07'
+PNA+PAT+7000000007:OPI+++SU:MILLER+FO:WINDY+MI:BRIAN+TI:MR'
+DTM+329:19460918:102'
+PDI+1'
+S10+10'
+CIN+UN'
+FTX+CID+++MICROANEURYSM R EYE'
```
Comment (n-r-b) [inline src/intTest/resources/success_uat_data/NHS002/full_acceptance_without_nhsack.edifact.dat:26]: Removed a number of segments temporarily until required parsers have been updated

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddress.java (hunk 1 above)
Comment (n-r-b) [reply src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddress.java:66]: That's a nice idea, i'll take a look 😄

Comment (n-r-b) [reply src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddress.java:66]: Hmm that is a very good point! 🙈

**Review by andrewwa-kainos — DISMISSED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddress.java (hunk 1 above)
Comment (n-r-b) [reply src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddress.java]: I've updated the above to use the `arrayGetSafe` method

**Review by andrewwa-kainos — APPROVED** [review]

File: src/intTest/resources/success_uat_data/NHS002/full_acceptance_without_nhsack.1.fhir.json [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      "https://fhir.nhs.uk/STU3/StructureDefinition/ITK-Message-Bundle-1"
+    ]
+  },
+  "identifier": {
+    "system": "https://tools.ietf.org/html/rfc4122",
+    "value": "26ed0510-a157-41b3-b5c7-71088575fb86"
+  },
+  "type": "message",
+  "entry": [
+    {
```
Comment (saileshp-kainos) [inline src/intTest/resources/success_uat_data/NHS002/full_acceptance_without_nhsack.1.fhir.json:16]: Is there meant to be an `Organization` FHIR resource here? 

The corresponding EDIFACT has two `NAD+SLA` segments (a `DEPT` and `ORG`).

Comment (antonlebedjko) [reply src/intTest/resources/success_uat_data/NHS002/full_acceptance_without_nhsack.1.fhir.json:16]: There is one, it's just below. Line 53

**Review by andrewwa-kainos — APPROVED** [review]

**Review by antonlebedjko — APPROVED** [review]
LGTM

**Review by saileshp-kainos — APPROVED** [review]
LGTM 🥳

**Review by andrewwa-kainos — APPROVED** [review]

**Review by antonlebedjko — APPROVED** [review]

**Pull request merged**